reviews/models/mixins.py

```diff
@@ -1,61 +1,118 @@
+    """Something that may be included in a reviewed collection and is subject to a reviews workflow.
     """
 
-    reviews_state = models.CharField(max_length=15, db_index=True, choices=States.choices(), default=States.PENDING.value)
-
-    def __init__(self, *args, **kwargs):
-        super(ReviewableMixin, self).__init__(*args, **kwargs)
+    __machine = None
 
-        self.__machine = ReviewsMachine(self)
+    # NOTE: reviews_state should rarely/never be modified directly -- use the state transition methods below
+    reviews_state = models.CharField(max_length=15, db_index=True, choices=workflow.States.choices(), default=workflow.States.INITIAL.value)
 
-    def reviews_accept(self, *args, **kwargs):
-        return self.__machine.accept(*args, **kwargs)
+    @property
+    def in_public_reviews_state(self):
+        public_states = workflow.PUBLIC_STATES.get(self.provider.reviews_workflow)
+        if not public_states:
+            return False
+        return self.reviews_state in public_states
+
+    def reviews_submit(self, user):
+        """Run the 'submit' state transition and create a corresponding ReviewLog.
+
+        Params:
+            user: The user triggering this transition.
+        """
+        with transaction.atomic():
+            self._reviews_machine.submit(user=user)
+            return self._reviews_machine.review_log
+
+    def reviews_accept(self, user, comment):
+        """Run the 'accept' state transition and create a corresponding ReviewLog.
+
+        Params:
+            user: The user triggering this transition.
+            comment: Text describing why.
+        """
+        with transaction.atomic():
+            self._reviews_machine.accept(user=user, comment=comment)
+            return self._reviews_machine.review_log
+
+    def reviews_reject(self, user, comment):
+        """Run the 'reject' state transition and create a corresponding ReviewLog.
+
+        Params:
+            user: The user triggering this transition.
+            comment: Text describing why.
+        """
+        with transaction.atomic():
+            self._reviews_machine.reject(user=user, comment=comment)
+            return self._reviews_machine.review_log
+
+    def reviews_edit_comment(self, user, comment):
+        """Run the 'edit_comment' state transition and create a corresponding ReviewLog.
+
+        Params:
+            user: The user triggering this transition.
+            comment: New comment text.
+        """
+        with transaction.atomic():
+            self._reviews_machine.edit_comment(user=user, comment=comment)
+            return self._reviews_machine.review_log
 
-    def reviews_reject(self, *args, **kwargs):
-        return self.__machine.reject(*args, **kwargs)
+    @property
+    def _reviews_machine(self):
+        if self.__machine is None:
+            self.__machine = ReviewsMachine(self, 'reviews_state')
+        return self.__machine
 
     class Meta:
```

Comment:
Class level variables should go above methods and properties.

The ideal class layout (IMO) is:

```python
class ClassName(superclass):

    class_level_variable = 'value

    objects = customManager()

    # If this is a model
    class Meta:
        table_name = 'something'

    field = models.TextField()

    @classmethod
    def class_level_methods(cls, args):
        pass

     @property
     def instance_properties(self):
        pass

    @instance_properties.setter
    def instance_properties(self, value):
        pass  # setters directly below

    def __init__(self):
        self.instance_var = None

    def instance_methods(self):
        pass

```

If you're placing it there because the OSF does that in some places, they're wrong. It should go at above everything else.